app-libs/stf/src/lib.rs

```diff
@@ -42,6 +47,24 @@ pub type BalanceTransferFn = ([u8; 2], AccountId, Compact<u128>);
 
 pub type ShardIdentifier = H256;
 
+pub type StfResult<T> = Result<T, StfError>;
+
+#[derive(Debug, Display, PartialEq, Eq)]
+pub enum StfError {
+	#[display(fmt = "Insufficient privileges {:?}, are you sure you are root?", _0)]
+	MissingPrivileges(AccountId),
+	#[display(fmt = "Error dispatching runtime call. {:?}", _0)]
+	Dispatch(String),
+	#[display(fmt = "Not enough funds to perform operation")]
+	MissingFunds,
+	#[display(fmt = "Account does not exist {:?}", _0)]
+	InexistentAccount(AccountId),
+	#[display(fmt = "Invalid Nonce {:?}", _0)]
+	InvalidNonce(Index),
+	StorageHashMismatch,
+	InvalidStorageDiff,
+}
```

Comment:
Moved the result and error type definition out of the `sgx`-only module, in order to have more parts of the code available in both `std` and `sgx` feature.